mypy/semanal.py

```diff
@@ -865,6 +865,20 @@ def analyze_func_def(self, defn: FuncDef) -> None:
                     return
                 assert isinstance(result, ProperType)
                 if isinstance(result, CallableType):
+                    # type guards need to have a positional argument, to spec
```

Comment:
What about pos-only?

Reply:
I think this works already, will add a test though.